### ISSUE TITLE
v1.8 - @var doesn't register a default string

### ISSUE THREAD

**Issue by AshleyNex** [Nxium-Developments/NextLanguage#10]
When trying to output a string using ``@output``, I discovered that default strings doesn't get registered at all in [parseVariable.js](https://github.com/Nxium-Developments/NextLanguage/blob/main/NextLanguage-Source/modules/parseVariable.js). The part where the string is suposed to be registered is not written anywhere. In the code I provided below shows where the code for the default string should've been.

This is the code that registers variable types in [parseVariable.js](https://github.com/Nxium-Developments/NextLanguage/blob/main/NextLanguage-Source/modules/parseVariable.js), but if you read the ``(type.toLowerCase() === "negative")``, all the way down to the bottom, you'll see that there's no registry for a default variable, which should've been a string.

```javascript
let parsedValue = value.trim();
    if (type.toLowerCase() === "negative") {
        parsedValue = parseFloat(value.replace("(negative)", "-"));
    } else if (type.toLowerCase() === "percent" || type.toLowerCase() === "percentage") {
        parsedValue = parseFloat(value.replace("%", ""));
    } else if (type.toLowerCase() === "int" || type.toLowerCase() === "integer") {
        parsedValue = parseInt(value, 10);
    } else if (type.toLowerCase() === "double") {
        parsedValue = parseFloat(value);
    } else if (type.toLowerCase() === "boolean") {
        parsedValue = value.toLowerCase() === "true";
    }
```
This is what a correct registry of all the variable types should've looked like. Noticed the 6th registry? It's the what's suposed to be in [parseVariable.js](https://github.com/Nxium-Developments/NextLanguage/blob/main/NextLanguage-Source/modules/parseVariable.js). But it isn't.

```javascript
let parsedValue = value.trim();
    if (type.toLowerCase() === "negative") {
        parsedValue = parseFloat(value.replace("(negative)", "-"));
    } else if (type.toLowerCase() === "percent" || type.toLowerCase() === "percentage") {
        parsedValue = parseFloat(value.replace("%", ""));
    } else if (type.toLowerCase() === "int" || type.toLowerCase() === "integer") {
        parsedValue = parseInt(value, 10);
    } else if (type.toLowerCase() === "double") {
        parsedValue = parseFloat(value);
    } else if (type.toLowerCase() === "boolean") {
        parsedValue = value.toLowerCase() === "true";
    } else if (type.toLowerCase() === "string") {
        parsedValue = value;
    }
```
This is what a correct registered variable type for a string should look like. It's not the set default type. But it works using this example code: ``@var [user]: (string)Someone``.

```javascript
if (type.toLowerCase() === "string") {
        parsedValue = value;
}
```
If we want to go even further and make the default variable type, be a string. The 7th line of code at the bottom which is ``else { parsedValue = value; type = "string"; }``. Basically makes whatever value, that is registered a string. So you could basically create new variable types to go along with.

```javascript
let parsedValue = value.trim();
    if (type.toLowerCase() === "negative") {
        parsedValue = parseFloat(value.replace("(negative)", "-"));
    } else if (type.toLowerCase() === "percent" || type.toLowerCase() === "percentage") {
        parsedValue = parseFloat(value.replace("%", ""));
    } else if (type.toLowerCase() === "int" || type.toLowerCase() === "integer") {
        parsedValue = parseInt(value, 10);
    } else if (type.toLowerCase() === "double") {
        parsedValue = parseFloat(value);
    } else if (type.toLowerCase() === "boolean") {
        parsedValue = value.toLowerCase() === "true";
    } else if (type.toLowerCase() === "string") {
        parsedValue = value;
    } else {
        parsedValue = value;
        type = "string";
    }
```

**Comment by AshleyNex** [comment]
Right, there's a flaw in my code. When trying to set one of the values in ``const [, name, type, value] = match;``, which is the **type** value, it outputs this error. Which just says that I can't set that string to a constant variable.

```
C:\Users\jedik\Documents\GitHub\Project Next\NextLanguage\NextLanguage-Source\modules\parseVariable.js:26
        type = "string";
             ^

TypeError: Assignment to constant variable.
    at parseVariable (C:\Users\jedik\Documents\GitHub\Project Next\NextLanguage\NextLanguage-Source\modules\parseVariable.js:26:14)
    at run (C:\Users\jedik\Documents\GitHub\Project Next\NextLanguage\NextLanguage-Source\patches\v1.8-rebulid.js:92:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

My update is to use ``let`` to change the type value into a non constant variable, which allows me to set the
variable type to a string, also update I was wrong, you can't create variable types along with the code I previously sent. Anything that isn't registered inside the variable types registry below, will now just default to a string, instead of crashing.
```javascript
const [, name, type, value] = match;

    let parsedValue = value.trim();
    let parsedtype = type.trim();
    if (type.toLowerCase() === "negative") {
        parsedValue = parseFloat(value.replace("(negative)", "-"));
    } else if (type.toLowerCase() === "percent" || type.toLowerCase() === "percentage") {
        parsedValue = parseFloat(value.replace("%", ""));
    } else if (type.toLowerCase() === "int" || type.toLowerCase() === "integer") {
        parsedValue = parseInt(value, 10);
    } else if (type.toLowerCase() === "double") {
        parsedValue = parseFloat(value);
    } else if (type.toLowerCase() === "boolean") {
        parsedValue = value.toLowerCase() === "true";
    } else if (type.toLowerCase() === "string") {
        parsedValue = value;
    } else {
        parsedValue = value;
        parsedtype = "string";
    }
```

**Comment by AshleyNex** [comment]
Fixed on v1.9
